ftl/code/controllers/subsystem/ftl_navigation.dm

```diff
@@ -0,0 +1,220 @@
+#define FTL_IDLE 0
+#define FTL_SPOOLUP 1
+#define FTL_JUMPING 2
+
+SUBSYSTEM_DEF(ftl_navigation)
+	name = "FTL Navigation"
+	wait = 10
+	init_order = INIT_ORDER_FTL_NAV
+	flags = SS_NO_FIRE
+
+	var/ftl_state = FTL_IDLE //is the ship in transit, spooling or idle?
+
+	var/datum/sector/current_sector
+	var/datum/sector/to_sector
+
+	var/list/visited_sectors = list()
+
+	var/sector_count = 0
+
+	var/list/ship_areas = list()
+
+	var/list/sector_name_fragments = list("le", "xe", "ge", "za", "ce", "bi", "so", "us", "es", "ar", "ma", "in", "di", "re", "a", "er", "at", "en", "be", "ra", "la", "ve", "ti", "ed", "or", "gu", "an", "te", "is", "ri", "on")
+
+	var/list/sector_types = list()
+	var/list/planet_types = list()
+
+	var/jump_knockdown_force = 3
+	var/jump_throw_force = 0 //Badmin var
+
+
+/datum/controller/subsystem/ftl_navigation/Initialize(timeofday)
+	for(var/A in get_areas(/area/, TRUE)) //Get all areas owned by the ship
+		var/area/place = A
+		if(place.z != 2) continue
+		ship_areas += place
+
+	for(var/type in subtypesof(/datum/planet)) //Get all planet types and weights
+		var/datum/planet/P = new type
+		planet_types += type
+		planet_types[type] = P.frequency

```

Comment:
Also last year ninja discovered you dont have to init these. Remove the first line that makes the instance and change p.frequency to initial[type.frequency] do the same for the others

Reply:
This change block possibility of random frequency in New(). I dont know, but random frequency make uneverse more randomly and interresting,,, may be.

Reply:
@Dennok If you wanted to have random frequency you could just do it in this for loop